share/completions/p4.fish

```diff
@@ -0,0 +1,726 @@
+# Fish completions for Perforce (p4) RCS
+# Based on the list of commands from
+# https://www.perforce.com/perforce/r16.1/manuals/cmdref/
+# Author: Boris Aranovich https://github.com/nomaed
+
+#########################################################
+### p4 command parsing
+#########################################################
+
+function __fish_print_p4_client_name
+    set -l p4info (p4 info -s 2> /dev/null)
+    if test -e "$p4info"
+        return
+    end
+    if string match -qr '^Client unknown'
+        return
+    end
+    string match -r '^Client name: .+' $p4info | string replace 'Client name: ' ''
+end
+
+#########################################################
+
+function __fish_print_p4_user_name
+    set -l p4info (p4 info 2> /dev/null)
+    if test -e "$p4info"
+        return
+    end
+    string match -r '^User name: .+' $p4info | string replace 'User name: ' ''
+end
+
+#########################################################
+
+function __fish_print_p4_changelists -d "Used 'detailed!' as first argument to use username@workspace prefix"
+    set -l detailed
+    if test -n "$argv"
+        and test $argv[1] = "detailed!"
+        set detailed true
+        set -e argv[1]
+    end
+
+    set -l changes (p4 changes -L $argv)
+
+    set -l result
+    for line in $changes
+        set line (string trim $line)
+        if test (string length $line) = 0
+            continue
+        end
+        set -l change_match (string match -ar '^Change ([0-9]+) on [0-9/]+ by (\S+).*$' $line)
+        if test -n "$change_match"
+            if test -n "$result"
+                echo $result
+                set result
+            end
+            set result $change_match[2]\t
+            if test "$detailed"
+                set result $result $change_match[3]:
+            end
+        else
+            set result $result $line
+        end
+    end
+
+    if test -n "$result"
+        echo $result
+    end
+end
+
+#########################################################
+
+function __fish_print_p4_opened_files
+    # p4 opened -s | sed 's/#.*//' | p4 -x - where | awk '/^\// {print $3}'
+    # p4 opened -s | string replace -ar '(^\S+).*$' '$1' | p4 -x - where | string replace -ar '\S+\s\S+\s(\S+)' '$1'
+    string replace -a "$PWD/" '' (p4 diff -sa -sb -sr)
+end
+
+#########################################################
+
+function __fish_print_p4_branches
+    set -l branches (p4 branches)
+    for branch in $branches
+        set -l matches (string match -ar '^Branch\s+(\S+)[^\']+\'(.+)\'$' $branch)
+        set -l cnt (count $matches)
+        if test (count $matches) -lt 2
+            continue
+        end
+        set -e matches[1]
+        echo -n $matches[1]
+        if test (count $matches) -gt 1
+            echo -n \t$matches[2]
+        end
+        echo
```

Comment:
Because the previous `echo`s in the loop are `echo -n` that do not output a `\n`, and the function is supposed to print a new line for every iteration.